### PR TITLE
Add code coverage for ToolStripItemCollection

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related #10453

Proposed changes
Add unit tests for ToolStripItemCollection.cs to test its properties & methods & events
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12316)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs [hunk 1]
```diff
@@ -120,4 +124,240 @@ public void ToolStripItemCollection_AddRange_ToolStripItemCollection_Success()
...
+    [WinFormsFact]
+    public void ToolStripItemCollection_MoveItem_Index_ItemWithParent_MovesItem()
+    {
+        using ToolStrip toolStrip1 = new();
+        using ToolStrip toolStrip2 = new();
+        ToolStripItem[] items = CreateToolStripItems(("Item1", "key1"), ("Item2", "key2"));
+        toolStrip1.Items.AddRange(items);
+
+        ToolStripItemCollection collection = new(toolStrip2, Array.Empty<ToolStripItem>());
+        collection.MoveItem(0, items[0]);
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs:328]: `collection.MoveItem(index: 0, items[0]);`

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs [hunk 2]
```diff
@@ -120,4 +124,240 @@ public void ToolStripItemCollection_AddRange_ToolStripItemCollection_Success()
...
+        ToolStripItem[] items = CreateToolStripItems(("Item1", "key1"), ("Item2", "key2"));
+
+        ToolStripItemCollection collection = new(toolStrip, items);
+
+        collection.Count.Should().Be(2);
+        collection[0].Text.Should().Be("Item1");
+        collection[1].Text.Should().Be("Item2");
+    }
+
+    private ToolStripItemCollection CreateToolStripItemCollectionWithItems(params (string Name, string Key)[] items)
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs:146]: Tip: The general practice for key-value pairs is to put the key in front and the value in the back. Can you consider changing the order?

`private ToolStripItem[] CreateToolStripItems(params (string Key, string Name)[] items)`

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs [hunk 3]
```diff
@@ -120,4 +124,240 @@ public void ToolStripItemCollection_AddRange_ToolStripItemCollection_Success()
         Assert.Equal("b", contextMenuStrip.Items[1].Text);
         Assert.Equal("c", contextMenuStrip.Items[2].Text);
     }
+
+    private ToolStripItem[] CreateToolStripItems(params (string Name, string Key)[] items)
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs:128]: Just a small suggestion: Although there is no fixed agreed order for tuple arrays, for ease of understanding and use, can you consider putting the Key first and the Value at the end?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs [hunk 4]
```diff
@@ -120,4 +124,240 @@ public void ToolStripItemCollection_AddRange_ToolStripItemCollection_Success()
...
+
+        collection.MoveItem(1, items[2]);
+
+        collection.IndexOf(items[2]).Should().Be(1);
+        collection.IndexOf(items[0]).Should().Be(0);
+        collection.IndexOf(items[1]).Should().Be(2);
+    }
+
+    [WinFormsFact]
+    public void ToolStripItemCollection_MoveItem_Index_ItemWithParent_MovesItem()
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs:320]: The title is not clear, please consider
`ToolStripItemCollection_MoveItem_MovesItemFromToolStripToCollection`

**Review by LeafShi1 — DISMISSED** [review]

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs [hunk 5]
```diff
@@ -120,4 +124,238 @@ public void ToolStripItemCollection_AddRange_ToolStripItemCollection_Success()
         Assert.Equal("b", contextMenuStrip.Items[1].Text);
         Assert.Equal("c", contextMenuStrip.Items[2].Text);
     }
+
+    private ToolStripItem[] CreateToolStripItems(params (string Key, string Name)[] items) =>
+        items.Select(item => new ToolStripButton(item.Name) { Name = item.Key }).ToArray();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs:129]: Please make sure to dispose the ToolStripButtons

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs [hunk 6]
```diff
@@ -120,4 +124,247 @@ public void ToolStripItemCollection_AddRange_ToolStripItemCollection_Success()
         Assert.Equal("b", contextMenuStrip.Items[1].Text);
         Assert.Equal("c", contextMenuStrip.Items[2].Text);
     }
+
+    private ToolStripItem[] CreateToolStripItems(params (string Key, string Name)[] items)
+    {
+        List<ToolStripItem> toolStripItems = new();
+        foreach (var item in items)
+        {
+            using ToolStripButton toolStripButton = new(item.Name) { Name = item.Key };
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs:133]: `using` would destroy the button when exiting the scope on line 135, we don't want it, it should stay alive until the test is run.

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemCollectionTests.cs:133]: But my original comment was wrong, we are always adding items to a toolstrip and then we dispose the toolstrip. Items will be disposed when the toolstrip is. Sorry about it.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Pull request merged**